vaporetto/src/ngram_model.rs

```diff
@@ -2,13 +2,20 @@ use alloc::vec::Vec;
 
 use bincode::{Decode, Encode};
 
-#[derive(Clone, Decode, Encode)]
+#[derive(Clone, Debug, Decode, Encode)]
 pub struct NgramData<T> {
     pub(crate) ngram: T,
     pub(crate) weights: Vec<i32>,
 }
 
-#[derive(Default, Decode, Encode)]
-pub struct NgramModel<T> {
-    pub(crate) data: Vec<NgramData<T>>,
+#[derive(Default, Debug, Decode, Encode)]
+pub struct NgramModel<T>(pub Vec<NgramData<T>>);
+
+#[derive(Clone, Debug, Decode, Encode)]
+pub struct TagNgramData<T> {
+    pub(crate) ngram: T,
+    pub(crate) weights: Vec<(u8, Vec<i32>)>,
```

Comment:
It would be helpful to implement this tuple using a struct with meaningful member names, e.g., `struct { offset, weight }`.

Reply:
4005607